app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java

```diff
@@ -25,67 +26,73 @@
 
 public class APIManager {
 
-    private boolean cacheEnabled;
+    public static final int MAX_COUNTRY_LIST_SIZE = 10;
 
-    private boolean longTermStorageEnabled;
+    public static final int MAX_GET_DATA_WORLD_CACHE_OLDNESS = 15; //time-unit is minutes
 
-    public static final int MAX_COUNTRY_LIST_SIZE = 10;
+    public static final int MAX_LIVE_STATISTICS_CACHE_OLDNESS = 15; //time-unit is minutes
+
+    public static final int MAX_CACHED_STATISTIC_CALLS = 50;
+
+    private static boolean cacheEnabled;
 
-    private final ExecutorService service;
+    private static boolean longTermStorageEnabled;
 
-    public APIManager(boolean cacheEnabled, boolean longTermStorageEnabled){
-        this.cacheEnabled = cacheEnabled;
-        this.longTermStorageEnabled = longTermStorageEnabled;
-        service = ThreadPoolHandler.getInstance();
+    private static final String TAG = APIManager.class.getName();
+
+    private static final ExecutorService service = ThreadPoolHandler.getInstance();
+
+    public static void setSettings(boolean cacheEnabled, boolean longTermStorageEnabled) {
+        APIManager.cacheEnabled = cacheEnabled;
+        APIManager.longTermStorageEnabled = longTermStorageEnabled;
     }
 
-    public List<Country> getDataWorld(API api) throws Throwable {
-        Logger.logD("APIManager.getDataWorld","Getting Data for every Country from api "+api.getName());
+    //gets the data of the whole world through the specified api -> throws the cause of the ExecutionException
+    public static List<Country> getDataWorld(API api) throws ExecutionException, JSONException, InterruptedException {
```

Comment:
you use async calls give back a Future instead of forcing it to be sync

Reply:
Looking into it now

Reply:
Doesn't yield a whole lot of performance and is just unneccessary complicating the code. Hence it will not be done...